dataset/dataset.cpp

```diff
@@ -11,12 +11,28 @@
 
 namespace scipp::dataset {
 
+namespace {
+template <class T> void expectWritable(const T &dict) {
```

Comment:
Could this function and the one in DataArray be made into a common function since it is a templated function anyway? Maybe you can then also refactor the function in `variable/variable.cpp`?

Reply:
I didn't feel it was worth it, since we need specific exception types and texts. Not much common.